ADR/0032-decoupling-deployment.md

```diff
@@ -0,0 +1,211 @@
+    NSTierTemplate --> |provides| ReleasePlan
+    A[fa:fa-code-commit Git Commit] --> |webhook| PipelinesAsCode
+    PipelinesAsCode --> |creates| build[Build PipelineRun]
+    build --> |triggers| integration-service
+    integration-service --> |asks| decision{Is there a ReleasePlan?}
+    ReleasePlan --> |influences| decision
+    decision --> |yes| Release
+    build --> |pushes the image| quay.io/redhat-user-workloads/tenant/component
+    Release --> |tags the image for release| quay.io/redhat-user-workloads/tenant/component
+    quay.io/redhat-user-workloads/tenant/component --> renovatebot[User provided RenovateBot]
+    renovatebot --> |updates| gitops-repo[User provided GitOps repo]
+    gitops-repo --> |causes deployment via| argocd[User provided ArgoCD]
+    argocd --> |deploys to| cluster[User provided cluster]
+```
+
+Outside of the AppStudio member cluster, the user is responsible for acquiring a gitops repo and
+deployment environments of their choice, manually laying out their application resources in the repo
+(assisted by tools like `kam`), specifying image references by tag to match the `:released` or
+`:validated` tagging schem mentioned above, configuring Argo to deploy from their gitops repo, and
+configuring [renovatebot] to propagate image updates by digest to their gitops repo.
+
+**For manual creation of new environments** - the user manages this directly using a combination of
+their gitops repo and argo, outside of the AppStudio member cluster.
+
+**For automated testing in ephemeral environments** - the user specifies an
+[IntegrationTestScenario] CR with a [DeploymentTargetClass] reference. After a build completes, but
+before it executes tests, the integration-service creates a new [DeploymentTargetClaim] CR with
+a [DTCLS] matching the one on the [IntegrationTestScenario]. This should cause the
+[deployment-target-operator] to provision and bind a [DeploymentTarget] matching the
+[DeploymentTargetClass]. [integration-service] then passes the kubeconfig and [Snapshot] to the
+integration test to deploy the user's app and run tests.  The integration-service should delete the
+[DTC] once it isn’t needed anymore for the test.
+
+```mermaid
+flowchart TD
+    User --> |provides| IntegrationTestScenario
+    IntegrationTestScenario --> |references| DeploymentTargetClass
+    commit[fa:fa-code-commit Git Commit] --> |webhook| PipelinesAsCode
+    PipelinesAsCode --> |creates| build[Build PipelineRun]
+    build --> |triggers| integration-service
+    integration-service --> |consults| IntegrationTestScenario
+    integration-service --> |creates| DeploymentTargetClaim
+    DeploymentTargetClaim --> |references| DeploymentTargetClass
+    DeploymentTargetClaim --> |is bound to| DeploymentTarget
+    DeploymentTarget --> |is associated with| kubeconfig[kubeconfig Secret]
+    kubeconfig --> |is provided to| test
+    integration-service --> |creates| test[Test PipelineRun]
+    DeploymentTarget --> |represents| cluster
```

Comment:
It can represent a namespace as well (and maybe an AMI instance in the future).

Reply:
Updated in 382bcb9.